rclcpp/test/rclcpp/test_subscription.cpp

```diff
@@ -461,6 +461,33 @@ TEST_P(TestSubscriptionInvalidIntraprocessQos, test_subscription_throws) {
   }
 }
 
+/*
+   Testing subscription with invalid use_intra_process_comm
+ */
+TEST_P(TestSubscriptionInvalidIntraprocessQos, test_subscription_throws_intraprocess) {
+  // Adds more coverage
+  rclcpp::SubscriptionOptionsWithAllocator<std::allocator<void>> options;
+  options.use_intra_process_comm = static_cast<rclcpp::IntraProcessSetting>(5);
+
+  initialize();
+  rclcpp::QoS qos = GetParam().qos;
+  using test_msgs::msg::Empty;
+  {
+    auto callback = std::bind(
+      &TestSubscriptionInvalidIntraprocessQos::OnMessage,
+      this,
+      std::placeholders::_1);
+
+    ASSERT_THROW(
```

Comment:
If we use the `RCLCPP_EXPECT_THROW_EQ` macro here, we can ensure that not only is it a `std::runtime_error`, but it also has the error message that we expect.  That ensures that this test is actually testing what we think it is, and not some other error.